From: Alec Leamas <leamas.alec@gmail.com>
Date: Mon, 20 Jul 2020 11:09:45 +0200
Subject: Build: Fix cmake missing NDEBUG flag (#1991).

Commands to test for the need for NDEBUG needs to run in sequence,
which cmake's execute_process does not guarantee.

Bug: https://github.com/OpenCPN/OpenCPN/issues/1991
Applied-Upstream: https://github.com/OpenCPN/OpenCPN/commit/fe56c2766b50e0
---
 CMakeLists.txt | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e145100..2cde27d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -792,16 +792,16 @@ if (GTK3_FOUND)
       printf("%d\\n", wxMAJOR_VERSION * 10 + wxMINOR_VERSION);
     }
   ]=])
+  set (TRY_COMPILE
+    "echo '${_VERSPROG}' > wx-32.c; cc $(wx-config --cflags) wx-32.c; ./a.out")
   execute_process(
-    COMMAND "/bin/sh" "-c" "echo '${_VERSPROG}' > wx-32.c"
-    COMMAND "/bin/sh" "-c" "cc $(wx-config --cflags) wx-32.c"
-    COMMAND "./a.out"
+    COMMAND "/bin/sh" "-c" "${TRY_COMPILE}"
     WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
     OUTPUT_VARIABLE _WX_VERSION
     ERROR_VARIABLE _WX_ERRORS
     OUTPUT_STRIP_TRAILING_WHITESPACE
   )
-  if (NOT "${_WX_ERRORS}" STREQUAL "")
+  if (NOT "${_WX_ERRORS}" STREQUAL "" OR "${_WX_VERSION}" STREQUAL "")
     message(FATAL_ERROR "Cannot run wxWidgets  version test (!)")
   elseif ("${_WX_VERSION}" LESS 32)
     add_definitions("-DNDEBUG")
